### DISCUSSION TITLE
Consuming a nuget package with native libraries in xunit project and .net framework

### DISCUSSION THREAD

**Discussion by DmitryLukyanov** [xunit/xunit#2280] [General]
Hey, I'm trying to consume our NuGet with native C libraries (MongoDB.Driver => dotnet add package MongoDB.Driver --version 2.12.1) in xunit test that is created in SDK csproj style and with .NET Framework 452. 

I see that in this case, the NuGet doesn't handle the native libraries and, as a result, tests fails. I suspect that this may be a xunit problem because this happens only in the test project (and only for .net framework TFs).
So if I make the same test with console application regardless TF, everything works correctly, the same correct handling happens if I set any core framework (I checked `netcoreapp11`, `netcoreapp21`, `netcoreapp31`, `net5`). 
Also, the behavior is not dependent on `shadow copy`, the used path is changed but the libraries still have not been copied.
Reproducing:
1. Create a xunit app in SDK csproj format. Manually replace the default TF on `<TargetFramework>net452</TargetFramework>`.
2. Consume MongoDB.Driver with the latest version: `dotnet add package MongoDB.Driver --version 2.12.1`
3. Try to run this test:

        [Fact]
        public void Test1()
        {
            _ = MongoDB.Libmongocrypt.Library.Version;
        }

Expected behavior: Test passing
Actual behavior: 

     System.IO.FileNotFoundException : Could not find: mongocrypt.dll -- ...

NOTE: I tried to fix this via additional applying this properties to the consumer app (all together and one at time):
1.    < RuntimeIdentifiers >win;win-x64</ RuntimeIdentifiers >
2.    < PlatformTarget >x64</ PlatformTarget >
3.    < NuGetRuntimeIdentifier >win</ NuGetRuntimeIdentifier >

but it didn't help. Thanks in advance.

**Comment by bradwilson** [comment]
I don't believe this has anything to do with xUnit.net itself, but rather is a question for the build team at Microsoft (the one that owns `msbuild` and `dotnet build`) to determine how to ensure that your unmanaged dependencies get copied into the output folder during build.
